test/evp_extra_test.c

```diff
@@ -1019,6 +1021,26 @@ static int test_EC_priv_pub(void)
         goto err;
     }
 
+    /* Positive and negative testcase for EVP_PKEY_get_octet_string_param */
+    if (!TEST_int_eq(EVP_PKEY_get_octet_string_param(params_and_pub,
+                                                     OSSL_PKEY_PARAM_ENCODED_PUBLIC_KEY,
+                                                     buffer, sizeof(buffer), &len), 1)
+        || !TEST_int_eq(len, 65))
+        goto err;
+
+    if (!TEST_int_eq(EVP_PKEY_get_octet_string_param(params_and_pub,
+                                                     OSSL_PKEY_PARAM_ENCODED_PUBLIC_KEY,
+                                                     NULL, 0, &len), 1)
+        || !TEST_int_eq(len, 65))
+        goto err;
+
+    /* too-short buffer len */
+    if (!TEST_int_eq(EVP_PKEY_get_octet_string_param(params_and_pub,
+                                                     OSSL_PKEY_PARAM_ENCODED_PUBLIC_KEY,
+                                                     buffer, 10, &len), 0)
+        || !TEST_int_eq(len, 65))
```

Comment:
This test is not right,,,
If the above test was to magically pass (which it wont), then we would expect the returned len <= 10 if we passed in a max buffer of 10. If it returned 65 it would indicate we just trashed the buffer.

Reply:
> This test is not right,,, If the above test was to magically pass (which it wont), then we would expect the returned len <= 10 if we passed in a max buffer of 10. If it returned 65 it would indicate we just trashed the buffer.

This case means if we input a too-short buffer, this function will not change the value of  ```len``` (which was set to 65 in the previous testcase).
It's really confusing here, maybe we can set the value of ```len``` to 0 before this test.

```
int EVP_PKEY_get_octet_string_param()
{
    OSSL_PARAM params[2];
    int ret1 = 0, ret2 = 0;
......
   /* return 0 here */
    if ((ret1 = EVP_PKEY_get_params(pkey, params)))
        ret2 = OSSL_PARAM_modified(params);
   /* will not change out_len here */
    if (ret2 && out_len != NULL)
        *out_len = params[0].return_size;
    return ret1 && ret2;
}
```

Reply:
I do NOT think we should check the return size is assigned in case of failure. IMO this is implementation-defined behavior.

Reply:
Sounds reasonable.
There is no reason for the caller to check ```out_len``` if the function has returned an error value